geniza/corpus/models.py

```diff
@@ -449,11 +450,35 @@ def unidecode_tag(sender, instance, **kwargs):
         instance.name = unidecode(instance.name)
 
 
+class DocumentQuerySet(models.QuerySet):
+    def metadata_prefetch(self):
+        return (
+            # super()
+            # .get_queryset()
+            self.select_related("doctype")
+            .prefetch_related(
+                "tags",
+                "languages",
+                Prefetch(
+                    "textblock_set",
+                    queryset=TextBlock.objects.select_related(
+                        "fragment", "fragment__collection"
+                    ),
+                ),
+            )
+            .annotate(shelfmk_all=ArrayAgg("textblock__fragment__shelfmark"))
+            .order_by("shelfmk_all")
+            # .order_by("id")   # @TODO: Where does this go?
+            # .prefetch_related("secondary_languages", "log_entries")
```

Comment:
this one is needed for csv export but not for admin; secondary prefetch method we can call in the export code but skip in admin?